### ISSUE TITLE
Add functions to build UI elements

### ISSUE THREAD

**Issue by roipoussiere** [tidalcycles/strudel#579]
Some functions could be added to allow the user to create UI elements and interact with them in real time.

Imagine a live performance where one artist writes code to make fun sounds and builds a modular synth on stage while the other artist creates the music with it.

Some UI elements ideas:
- drop-down menu
- switches
- buttons
- matrix pad
- slider
- potentiometer
- drawing area
- microphone (to record sound samples)?

The API must be discussed, but I think it's important to both allow:
- state-based approach, ex. `.cutoff(my_slider.getValue())`;
- event-based aproach, ex. `my_button.on_pressed(btn => { s("bd") } )`.

I would love to try to work on it. :)

**Comment by felixroos** [comment]
yep this is a planned feature, see https://github.com/tidalcycles/strudel/discussions/353
implementing things directly in the repl is also a fun way to share / discuss ideas, for example https://strudel.tidalcycles.org/?STsblGtree5S
